app/src/App/Domain/Service/UsersService.php

```diff
@@ -36,7 +36,13 @@ public function getAll(): array
 
     public function getById(UserId $id): ?User
     {
-        return $this->users->findById($id);
+        $user = $this->users->findById($id);
+
+        if (!$user instanceof User) {
+            throw UserNotFoundException::fromUsername($username);
```

Comment:
since now we're throwing an exception, it should be the case to add a test for it